Fix bug in R_PCRE autoconf macro which leads to 'configure' losing '-lz'
and/or '-lbz2' from LIBS (under certain circumstances, and only relevant
if "system" versions of these libraries are used).  (cf. #18229)

--- R-3.3.1_original/m4/R.m4	2016-03-17 00:02:06.000000000 +0100
+++ R-3.3.1_patched/m4/R.m4	2016-08-20 10:17:08.663006152 +0200
@@ -3193,7 +3193,6 @@
 #endif
 }
 ]])], [r_cv_have_pcre810=yes], [r_cv_have_pcre810=no], [r_cv_have_pcre810=no])])
-fi
 if test "x${r_cv_have_pcre810}" != xyes; then
   have_pcre=no
   LIBS="${r_save_LIBS}"
@@ -3216,6 +3215,7 @@
 }
 ]])], [r_cv_have_pcre832=yes], [r_cv_have_pcre832=no], [r_cv_have_pcre832=no])])
 fi
+fi
 
 AC_MSG_CHECKING([whether PCRE support suffices])
 if test "x${r_cv_have_pcre810}" != xyes; then
